src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/dto/status/PyrisStatusUpdateDTO.java

```diff
@@ -0,0 +1,16 @@
+package de.tum.in.www1.artemis.service.connectors.pyris.dto.status;
+
+import java.util.List;
+
+public abstract class PyrisStatusUpdateDTO {
```

Comment:
please use records

Reply:
Since records do not support inheritance and this each sub-feature of Iris will have its own StatusUpdateDTO, we have to use classes here.
The alternative would be to duplicate the common attributes (currently only `stages`) for each specific StatusUpdateDTO.

The same also goes for `PyrisPipelineExecutionDTO`, which currently has 2 attributes that are shared for all implementation.

Which do you prefer?

Reply:
yes, we decided that duplicating common attributes for DTOs in records is perfectly fine and we do not care about missing inheritance. In fact records have so many advantages, that duplication is really not a problem here!

Reply:
advantages of records:
* no boilerplate code for constructor, getter, setter, equals, hashCode, toString
* record instances are immutable which leads to fewer error possibilities
* DTOs should NOT be reused for different REST calls, because this can be problematic. Records enforce you better to think about what data is actually needed in the communication and you start from scratch instead of adding all kind of attributes even if they are not necessary. Therefore, reuse is not desired here.
* ...

Reply:
Thanks a lot for the explanation here and on Slack! I kinda missed the additions to the docs for this topic 😅

I've now converted all DTOs of Pyris to records. ✅